lonboard/experimental/traits.py

```diff
@@ -101,3 +102,71 @@ def validate(
 
         self.error(obj, value)
         assert False
+
+
+class NormalAccessor(Union[TraitType, FixedErrorTraitType]):
+    """
+        A representation of a deck.gl normal accessor
+        
+        Acceptable inputs:
+        - A numpy ndarray with two dimensions: The size of the second dimension must be 3 i.e. `(N,3)`
+        - a pyarrow `FixedSizeListArray` or `ChunkedArray` containing `FixedSizeListArray`s 
+        where the size of the inner fixed size list 3.
+    """
+    default_value = [0,0,1]
+    info_text = (
+        "List representing normal of each object, in [nx, ny, nz]. or numpy ndarray or "
+        "pyarrow FixedSizeList representing the normal of each object, in [nx, ny, nz]"
+    )
+
+    def __init__(
+        self: TraitType,
+        *args,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.tag(sync=True, **NORMAL_SERIALIZATION)
+        
+    def validate(
+        self, obj, value
+    ) -> Union[Tuple[int, ...], List[int], pa.ChunkedArray, pa.FixedSizeListArray]:
+        
+        
+        # TODO Redundant logic?
+        list_size = value.shape[1]
+        if list_size != 3:
+            self.error(
+                obj,
+                value,
+                info="Point array must have 3 as its second dimension. I.e. (N,3)",
+            )
+        
+        if isinstance(value, np.ndarray):
+            # values must be contiguous (the same length for all values)
+            if value.ndim != 2 or value.shape[1] != 3:
+                self.error(obj, value, info="Normal array must be 2D with shape (N,3)")
+                
+            if not np.any(value == 1):
+                self.error(obj, value, info="One of the normal vector elements must be 1.")
+                
+            return pa.FixedSizeListArray.from_arrays(value.flatten("C"), list_size)
```

Comment:
We need to ensure that numpy arrays are floating point data type. You can check with `np.issubdtype(value.dtype, np.floating)` and then we want to cast it to float32 if it's not already.
```py
if not np.issubdtype(value.dtype, np.float32):
    value.astype(np.float32)
```